src/main/java/frc/trigon/robot/motorsimulation/MotorSimulation.java

```diff
@@ -0,0 +1,140 @@
+    private double voltageCompensationSaturation;
+    private MotorSimulationConfiguration config;
+    private double voltage = 0;
+
+    public MotorSimulation() {
+        REGISTERED_SIMULATIONS.add(this);
+    }
+
+    public void applyConfiguration(MotorSimulationConfiguration config) {
+        profiledPIDController = new ProfiledPIDController(config.pidConfigs.kP, config.pidConfigs.kI, config.pidConfigs.kD, new TrapezoidProfile.Constraints(config.motionMagicConfigs.maxVelocity, config.motionMagicConfigs.maxAcceleration));
+        pidController = new PIDController(config.pidConfigs.kP, config.pidConfigs.kI, config.pidConfigs.kD);
+        voltageCompensationSaturation = config.voltageCompensationSaturation;
+        this.config = config;
+        enablePIDContinuousInput(config.pidConfigs.enableContinuousInput);
+    }
+
+    public void stop() {
+        setVoltage(0);
+    }
+
+    public void setControl(VoltageOut voltageRequest) {
+        setVoltage(voltageRequest.Output);
+    }
+
+    public void setControl(PositionVoltage positionVoltageRequest) {
+        this.positionVoltageRequest = positionVoltageRequest;
+        motionMagicRequest = null;
+        double voltage = pidController.calculate(getPosition(), positionVoltageRequest.Position);
+        setVoltage(voltage);
+    }
+
+    public void setControl(MotionMagicVoltage motionMagicRequest) {
+        this.motionMagicRequest = motionMagicRequest;
+        positionVoltageRequest = null;
+        double pidOutput = profiledPIDController.calculate(getPosition(), motionMagicRequest.Position);
+        double feedforwardOutput = calculateFeedforward(
+                config.feedForwardConfigs,
+                Units.rotationsToRadians(motionMagicRequest.Position / config.conversionFactor),
+                profiledPIDController.getGoal().velocity
+        );
+        double output = pidOutput + feedforwardOutput;
+        setVoltage(output);
+    }
+
+    public void setControl(DutyCycleOut dutyCycleRequest) {
+        double voltage = Conversions.compensatedPowerToVoltage(dutyCycleRequest.Output, voltageCompensationSaturation);
+        setVoltage(voltage);
+    }
+
+    public double getVoltage() {
+        return voltage;
+    }
+
+    public double getPosition() {
+        return getPositionRevolutions() * config.conversionFactor;
+    }
+
+    public double getVelocity() {
+        return getVelocityRevolutionsPerSecond() * config.conversionFactor;
+    }
+
+    private static void updateRegisteredSimulations() {
+        for (MotorSimulation motorSimulation : REGISTERED_SIMULATIONS) {
+            motorSimulation.updateMotor();
+            if (motorSimulation.motionMagicRequest != null)
+                motorSimulation.setControl(motorSimulation.motionMagicRequest);
+            else if (motorSimulation.positionVoltageRequest != null)
+                motorSimulation.setControl(motorSimulation.positionVoltageRequest);
+        }
+    }
```

Comment:
Static methods should be at the top of the class. Look at the position in the current watcher.